sprokit/processes/core/merge_track_sets_process.cxx

```diff
@@ -0,0 +1,236 @@
+  }
+
+  std::map< vital::track_id_t, vital::track_id_t >& mappings = id_remapping[index];
+
+  for( auto track_ptr : input->tracks() )
+  {
+    if( !track_ptr )
+    {
+      continue;
+    }
+
+    const vital::track_id_t id = track_ptr->id();
+    vital::track_id_t mapped_id;
+    auto element = mappings.find( id );
+
+    if( element != mappings.end() )
+    {
+      mapped_id = element->second;
+    }
+    else
+    {
+      if( used_ids.find( id ) == used_ids.end() )
+      {
+        mapped_id = id;
+        highest_id = std::max( highest_id, id );
+      }
+      else
+      {
+        highest_id = highest_id + 1;
+        mapped_id = highest_id;
+      }
+
+      used_ids.insert( mapped_id );
+      mappings[ id ] = mapped_id;
+    }
+
+    track_ptr->set_id( mapped_id );
+    output->insert( track_ptr );
+  }
+}
+
+
+// ============================================================================
+
+merge_track_sets_process
+::merge_track_sets_process( vital::config_block_sptr const& config )
+  : process( config ),
+    d( new merge_track_sets_process::priv )
+{
+  make_ports();
+  make_config();
+}
+
+
+merge_track_sets_process
+::~merge_track_sets_process()
+{
+}
+
+
+// ----------------------------------------------------------------------------
+void merge_track_sets_process
+::_configure()
+{
+}
+
+
+// ----------------------------------------------------------------------------
+void
+merge_track_sets_process
+::_step()
+{
+  std::vector< vital::track_set_sptr > track_list;
+
+  for( const auto port_name : d->p_port_list )
+  {
+    vital::track_set_sptr track_sptr =
+      grab_from_port_as< vital::object_track_set_sptr >( port_name );
+
+    track_list.push_back( track_sptr );
+  }
+
+  if( track_list.size() > d->id_remapping.size() )
+  {
+    d->id_remapping.resize( track_list.size() );
+  }
+
+  // Merge tracks sequentially
+  vital::track_set_sptr output;
+
+  if( track_list.empty() )
+  {
+    LOG_WARN( logger(), "No input tracks provided" );
+  }
+
+  for( unsigned i = 0; i < track_list.size(); ++i )
+  {
+    d->add_tracks_to_set( track_list[i], i, output );
+  }
+
+  // Return by value
+  push_to_port_using_trait( object_track_set,
+    std::dynamic_pointer_cast< vital::object_track_set >( output ) );
+}
+
+
+// ----------------------------------------------------------------------------
+void merge_track_sets_process
+::make_ports()
+{
+  // Set up for required ports
+  sprokit::process::port_flags_t optional;
+  sprokit::process::port_flags_t required;
+  required.insert( flag_required );
+
+  // -- output --
+  declare_output_port_using_trait( object_track_set, required );
+}
+
+
+// ----------------------------------------------------------------------------
+void merge_track_sets_process
+::make_config()
+{
+
+}
+
+
+// ============================================================================
+sprokit::process::port_info_t
+merge_track_sets_process
+::_input_port_info(port_t const& port_name)
```

Comment:
Overriding this method will work, but there is a newer API (something like input_port_undefined() ) that does the same thing but is not called when data passes through a port to this process. It just of an efficiency thing.